src/answers-umd.js

```diff
@@ -726,6 +727,54 @@ class Answers {
       console.error(`Invalid visitor. Visitor was not set because "${JSON.stringify(visitor)}" does not have an id.`);
     }
   }
+
+  /**
+   * Initialize visibility change event listener(s) to send analytics
+   * events when user close or navigate away from a result page.
+   */
+  _initVisibilityChangeListeners (verticalKey) {
+    /**
+     * Safari desktop listener and IE11 listeners fire visibility change event twice when switch
+     * to new tab and then close browser. This variable is used to ensure RESULTS_HIDDEN analytics event
+     * does not get send again if the page is already hidden.
+     */
+    let documentVisibilityState;
+    document.addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'hidden' && documentVisibilityState !== 'hidden') {
+        documentVisibilityState = 'hidden';
+        this._reportVisibilityChangeEvent(verticalKey, 'RESULTS_HIDDEN');
+      } else if (document.visibilityState === 'visible') {
+        documentVisibilityState = 'visible';
+        this._reportVisibilityChangeEvent(verticalKey, 'RESULTS_UNHIDDEN');
+      }
+    });
+
+    /**
+     * VisibilityChange API does not register when page is terminated (close tab/browser) in IE11.
+     * So, unload event is used to capture those RESULTS_HIDDEN scenarios.
+     */
+    if (isIE()) {
+      window.addEventListener('unload', () => {
+        if (documentVisibilityState !== 'hidden') {
+          documentVisibilityState = 'hidden';
```

Comment:
do we need to update the documentVisibilityState if the page is about the be closed?

Reply:
I also feel like it could be nice if somebody closing the page is a different analytics dimension, I'm kinda curious if the spec says anything about that but eh

Reply:
are you suggesting another analytics event when page terminate instead of just in background instead? From the spec, I think the current plan with search duration calculations is just based on RESULTS_HIDDEN and RESULTS_UNHIDDEN only (and follow up query events) so they are treating hide and close page to be the same thing.

Reply:
Good point, if it's closing anyway, I guess we don't need to update documentVisibilityState. I will remove it